src/models/mSeriesChallenge.ts

```diff
@@ -0,0 +1,305 @@
+
+    if (options.includeOwnedParticipantData) {
+      const { mSeriesChallengeParticipant } = await this.action('mSeriesChallengeParticipant:show', { mSeriesChallengeId: this._mSeriesChallengeData.id }) as MSeriesChallengeParticipantResponse
+      this.setMSeriesChallengeParticipant(mSeriesChallengeParticipant)
+    }
+
+    this.setMSeriesChallenge(mSeriesChallenge)
+
+    return this
+  }
+
+  /**
+   * Method to end a challenge. Only challenge owners can end a challenge.
+   */
+  async end () {
+    await this.action('mSeriesChallenge:delete', { id: this._mSeriesChallengeData.id })
+  }
+
+  /**
+   * Method to leave a challenge. Only challenge participants can leave a challenge.
+   */
+  async leave () {
+    await this.action('mSeriesChallengeParticipant:delete', { mSeriesChallengeId: this._mSeriesChallengeData.id })
+  }
+
+  /**
+   * Method to join a challenge. Challenge instance must contain joinCode to successfully join.
+   */
+  async join () {
+    if (this._mSeriesChallengeData.joinCode !== undefined) {
+      const { mSeriesChallengeParticipant } = await this.action('mSeriesChallengeParticipant:create', { joinCode: this._mSeriesChallengeData.joinCode }) as MSeriesChallengeParticipantResponse
+      this.setMSeriesChallengeParticipant(mSeriesChallengeParticipant)
+      return mSeriesChallengeParticipant
+    }
+    return undefined
+  }
+
+  /**
+   * This method can return the current session user's participant data or a different user's participant data.
+   *
+   * @param mSeriesChallengeParticipantId Id of participant to be shown. Leave undefined for current session user's participant data.
+   * @returns An MSeries Challenge Participant
+   */
+  async getParticipant (mSeriesChallengeParticipantId?: number) {
+    if (mSeriesChallengeParticipantId !== undefined) {
+      const { mSeriesChallengeParticipant } = await this.action('mSeriesChallengeParticipant:show', { mSeriesChallengeParticipantId: mSeriesChallengeParticipantId }) as MSeriesChallengeParticipantResponse
+      return mSeriesChallengeParticipant
+    } else {
+      const { mSeriesChallengeParticipant } = await this.action('mSeriesChallengeParticipant:show', { mSeriesChallengeId: this._mSeriesChallengeData.id }) as MSeriesChallengeParticipantResponse
+      this.setMSeriesChallengeParticipant(mSeriesChallengeParticipant)
+      return mSeriesChallengeParticipant
+    }
+  }
+
+  /**
+   * Method to search for participants in challenge.
+   * Session User must either own or have joined challenge to search for other participants.
+   *
+   * @param options.nameSearchQuery Name to search for
+   * @param options.sort default: 'name'
+   * @param options.ascending default: true
+   * @param options.limit default: 20
+   * @param options.offset default: 0
+   * @returns An array of MSeries Challenge Participants
+   */
+  async searchParticipantsByName (options: {
```

Comment:
Shouldn't this just be a `getParticipants` with query options? "Search" isn't a verb we've previously used.